sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1561,6 +1562,129 @@ def plot_loading_results(self, subs=None):
+        l = self.length
+        C3 = Symbol('C3')
+        C4 = Symbol('C4')
+
+        shear_curve = limit(self.shear_force(), x, l) - val
+        moment_curve = limit(self.bending_moment(), x, l) - val*(l-x)
+
+        slope_eqs = []
+        deflection_eqs = []
+
+        slope_curve = integrate(self.bending_moment(), x) + C3
+        for position, value in self._boundary_conditions['slope']:
+            eqs = slope_curve.subs(x, position) - value
+            slope_eqs.append(eqs)
+
+        deflection_curve = integrate(slope_curve, x) + C4
+        for position, value in self._boundary_conditions['deflection']:
+            eqs = deflection_curve.subs(x, position) - value
+            deflection_eqs.append(eqs)
+
+        solution = list((linsolve([shear_curve, moment_curve] + slope_eqs
+                            + deflection_eqs, (C3, C4) + reactions).args)[0])
+        solution = solution[2:]
+        # Determining the equations and solving them.
+        self._ild_reactions = dict(zip(reactions, solution))
+        return self._ild_reactions
+
+    def plot_ild_reactions(self):
+        """
+
+        Plots the Influence Line Diagram of Reaction Forces
+        under the effect of a moving load. This function
+        should be called after calling solve_for_ild_reactions().
+
+        Examples
+        ========
+
+        There is a beam of length 10 meters. A point load of magnitude 5KN
+        is also applied from top of the beam, at a distance of 4 meters
+        from the starting point. There are two simple supports below the
+        beam, located at the starting point and at a distance of 7 meters
+        from the starting point. Plot the I.L.D. equations for reactions
+        at both support points under the effect of a moving load
+        of magnitude 1kN.
+
+        Using the sign convention of downwards forces being positive.
+
+        .. plot::
+            :context: close-figs
+            :format: doctest
+            :include-source: True
+
+            >>> from sympy import symbols
+            >>> from sympy.physics.continuum_mechanics.beam import Beam
+            >>> E, I = symbols('E, I')
+            >>> R_0, R_7 = symbols('R_0, R_7')
+            >>> b = Beam(10, E, I)
+            >>> b.apply_support(0, 'roller')
+            >>> b.apply_support(7, 'roller')
+            >>> b.apply_load(5,4,-1)
+            >>> b.solve_for_ild_reactions(1,R_0,R_7)
+            {R_0: x/7 - 22/7, R_7: -x/7 - 20/7}
+            >>> b.plot_ild_reactions()
+            PlotGrid object containing:
+            Plot[0]:Plot object containing:
+            [0]: cartesian line: x/7 - 22/7 for x over (0.0, 10.0)
+            Plot[1]:Plot object containing:
+            [0]: cartesian line: -x/7 - 20/7 for x over (0.0, 10.0)
+
+        """
+        if not self._ild_reactions:
+            raise KeyError("solve_for_ild_reactions() not called")
+
+        x = self.variable
+        p = []
+        for i in self._ild_reactions:
```

Comment:
Here too, it can be `for reaction, equation in self._ild_reactions` or something better than just `i`